lib/elixir/src/elixir_env.erl

```diff
@@ -85,10 +85,20 @@ reset_unused_vars(#elixir_ex{unused={_Unused, Version}} = S) ->
   S#elixir_ex{unused={#{}, Version}}.
 
 check_unused_vars(#elixir_ex{unused={Unused, _Version}}, E) ->
-  [elixir_errors:file_warn(Meta, E, ?MODULE, {unused_var, Name, Overridden}) ||
+  [elixir_errors:file_warn([calculate_span(Name, Meta) | Meta], E, ?MODULE, {unused_var, Name, Overridden}) ||
     {{{Name, nil}, _}, {Meta, Overridden}} <- maps:to_list(Unused), is_unused_var(Name)],
   E.
 
+calculate_span(Name, Meta) ->
+  Line = ?line(Meta),
+  case lists:keyfind(column, 1, Meta) of
+    {column, Column} ->
+      {span, {Line, Column + string:length(atom_to_binary(Name))}};
+
+    _ ->
+      []
```

Comment:
The empty list is not going to behave as you expect. In any case, since you already have the Meta here, I would make it so `calculate_span(Meta, Name)` returns the potentially updated meta.

Btw, this module is a good place to have this function, you can expose it from here. :)